src/drivers/abstract/index.js

```diff
@@ -1,18 +1,211 @@
 'use strict';
 
+const SqlString = require('sqlstring');
+
 const Logger = require('./logger');
+const Attribute = require('./attribute');
+const DataTypes = require('../../data_types');
+const Spellbook = require('./spellbook');
+const { heresql } = require('../../utils/string');
 
 /**
  * Migration methods
  * - https://dev.mysql.com/doc/refman/8.0/en/create-table.html
  * - https://dev.mysql.com/doc/refman/8.0/en/alter-table.html
  */
 
-module.exports = class AbstractDriver {
+class AbstractDriver {
+
+  // define static properties as this way IDE will prompt
+  static Spellbook = Spellbook;
+  static Attribute = Attribute;
+  static DataTypes = DataTypes;
+
   constructor(opts = {}) {
     const { logger } = opts;
     this.logger = logger instanceof Logger ? logger : new Logger(logger);
     this.idleTimeout = opts.idleTimeout || 60;
     this.options = opts;
+    this.Attribute = this.constructor.Attribute;
+    this.DataTypes = this.constructor.DataTypes;
+    this.spellbook = new this.constructor.Spellbook();
+    this.escape = SqlString.escape;
+    this.escapeId = SqlString.escapeId;
+  }
+
+  static #isLeoricDriver = true;
+
+  static isLeoricDriver() {
```

Comment:
I didn't find a proper way to tell one class is another class's subclass.

Reply:
You can check the inheritance with `SubClass.prototype instanceof Class`